File: src/hackerrank/java/PageCount.java
package hackerrank.java;

public class PageCount {

    public static void main(String[] args) {
        System.out.println(pageCount(4,4));
    }

    private static int pageCount(int n, int p){
        int left = p / 2;
        if(n%2==0){
            n++;
        }
        int right =(n-p) / 2;
        return left < right ? left : right;
    }
}

File: src/hackerrank/kotlin/PageCount.kt
package hackerrank.kotlin

fun main(args: Array<String>) {
    println(pageCount(6, 2))
}

fun pageCount(n: Int, p: Int): Int {

    var total = n
    if (n % 2 == 0){
        total++
    }
    val left = p/2
    val right = (total-p)/2

    return if (left< right) left else right

}